pysmt/rewritings.py

```diff
@@ -543,7 +546,7 @@ def walk_quantifier(self, formula, args, **kwargs):
 
     def walk_theory_op(self, formula, **kwargs):
         #pylint: disable=unused-argument
-        return None
+        return [],None
```

Comment:
FYI, @sureshgl,  this is way the prenex rewriter was not working on the examples you commented out.
In python it is ok to have different types for the return value. It is not beautiful, but in selected cases it can make your life easier. In this case, we have a special case to handle (theory formula), that we represent with None. If we return always a tuple, and only set the second element to None, every time we want to check the arguments of the call, we need to do unpacking. Here we know that the first argument is useless if the second is None, and simplify this to None.

I think that you changed this, because you were receiving some error on ```StrContains```, ```StrPrefixOf```, ```StrSuffixOf```. This is because these are not *only* theory operators, but predicates, i.e., they have a Boolean return value. This means that they need to be taken into account when manipulating the boolean structure of the formula.

I fixed it. We probably need to clarify this in the documentation of the rewriter.